tests/marray/marray_operators.h

```diff
@@ -224,6 +224,52 @@ class run_binary_sequence_scalar {
   }
 };
 
+template <typename InitSeqForRhs, std::size_t seq_el_num>
+inline constexpr bool init_seq_contains_too_big_values_for_shift_op(
+    std::size_t max_shift_wo_undef_behavior) {
+  return (std::is_same_v<InitSeqForRhs, seq_inc> ||
+          std::is_same_v<InitSeqForRhs, seq_dec<seq_el_num>>)&&seq_el_num >
+         max_shift_wo_undef_behavior;
+}
+
+/**
+ * @brief The function checks that for particular test case there is no
+ * undefined behavior for left or right shift operation. In case of shift
+ * operation too large value of right hand side argument can lead to undefined
+ * behavior. For initialization sequences: seq_inc, seq_dec the right hand side
+ * argument can take values up to 64. So we should exclude test cases for such
+ * big shift values. Left shift operation by 8 bits is guaranteed legal for our
+ * tests beacuse maximum value after shift operation will be 64 * 2^8 = 2^14 (64
```

Comment:
```suggestion
 tests because maximum value after shift operation will be 64 * 2^8 = 2^14 (64
```
PS : all these `*` does not bring anything to the clarity since we are not using  anymore https://en.wikipedia.org/wiki/Computer_programming_in_the_punched_card_era :-)
At least you could use a spell-checker instead. :-)

Reply:
We perform left shift operation for different values for lhs argument: 1,2...64 not only for values equal to 1. According to c++ specification left shift operation for signed types can lead to undefined behavior if the result value of the operation: `a << b == a * 2^b` can't be represented in unsigned version of return type. Values: `a << (sizeof(T)*CHAR_BITS - 1)` can't be represented in unsigned version of return type if lhs argument is grater than 1. Therefore we can't limit values of rhs argument for left shift operation only based on lhs argument type, we should take into account value of lhs argument. However such an approach is acceptable for `char` type, due to `char` values will be promoted to `int` before operation and `(sizeof(T)*CHAR_BITS - 1)` will be equal to 7 and even `64 << 7` can be represented in `unsigned int`. But for other types result of operation will be out of range of unsigned version of return type. Of course for different cases we can have values less than 64 for lhs argument and calculate acceptable values for rhs argument. But I thought it would make code more complex and I decided just to limit value for rhs argument to 8. It is acceptable for all cases in that test because of `int` type according to c++ specification should be at least 16 bits.

Removed all extra `*` from comment.